panels/applications/cc-applications-panel.c

```diff
@@ -1715,6 +1745,30 @@ cc_applications_panel_init (CcApplicationsPanel *self)
   self->privacy_settings = g_settings_new ("org.gnome.desktop.privacy");
   self->search_settings = g_settings_new ("org.gnome.desktop.search-providers");
 
+   /* FIXME: should become asynchronous */
+  system_bus = g_bus_get_sync (G_BUS_TYPE_SYSTEM, self->cancellable, &error);
+  if (system_bus == NULL)
+    {
+      g_warning ("Error getting system bus while setting up app permissions: %s", error->message);
+      return;
+    }
+
+  /* Load the user’s parental controls settings too, so we can filter the list. */
+  self->manager = mct_manager_new (system_bus);
+  self->app_filter = mct_manager_get_app_filter (self->manager,
```

Comment:
I don’t really like these sync calls in the constructor, but since g-c-c doesn’t set a better precedent anywhere else (that I know of), we’d end up doing a lot of work to make this async, which isn’t worth it at the moment.

So the sync wheel keeps turning, and the async one never gets started.

Reply:
Yeah, I agree very much to this. `_async`-fying as downstream-first, might makes the next rebase painful to some extend because code-bits coming from upstream will get moved a lot too as a result. Hence, not looking forward to it at all during rebases :(